enaml/core/import_hooks.py

```diff
@@ -368,26 +423,203 @@ def get_code(self):
+                The fully path to the archive.
+
+        Returns
+        -------
+        results : bool
+                Whether the archive is supported or not.
+        """
+        return os.path.splitext(archive_path)[-1].lower() in cls.supported_archives
+
+    def __init__(self, file_info, archive_path):
+        """ Initialize an importer object.
+
+        Parameters
+        ----------
+        file_info : EnamlFileInfo
+            An instance of EnamlFileInfo.
+        archive_path : String 
+            File path to the archive to import from.
+
+        """
+        super(EnamlZipImporter, self).__init__(file_info)
+        self.archive_path = archive_path
+        self.archive = None  # Reference to opened archive
+
+    def get_src_mod_time(self, src_path):
+        """ Overridden to read the modified time of the archive 
+        instead of the source file.
+        """
+        return super(EnamlZipImporter, self).get_src_mod_time(self.archive_path)
+
+    def read_source(self, src_path):
+        """ Overridden to read the source from the currently opened archive 
+        instead of the source file. The `self.archive` must be a reference
+        to the current archive object.
+        """
+        if sys.version_info.major > 2:
+            return self.archive.read(src_path).decode()
+        else:
+            return self.archive.read(src_path)
+
+    def get_code(self):
+        """ Loads and returns the code object for the Enaml module and
+        the full path to the module for use as the __file__ attribute
+        of the module.
+
+        Returns
+        -------
+        result : (code, path)
+            The Python code object for the .enaml module, and the full
+            path to the module as a string.
+
+        """
+
+        # Try to load from the cache on disk
+        cache = self.get_cached_code(self.archive_path)
+        if cache is not None:
+            return cache
+
+        # Otherwise load it from the archive
+        file_info = self.file_info
+        with ZipFile(self.archive_path, 'r') as archive:
+            # Path within the archive that should contain the cached module
+            code_cache_path = os.path.relpath(file_info.cache_path,
+                                              self.archive_path).replace("\\", "/")
+
+            # Try to use the cached file embedded in the archive
+            if code_cache_path in archive.namelist():
+                # Compile the cached code
+                cache = archive.read(code_cache_path)
+                code = marshal.loads(cache[8:])
+                return (code, code_cache_path)
+
+            #: Save reference
+            self.archive = archive
+
+            # Otherwise, compile from source and attempt to cache it on the system
+            code_path = os.path.relpath(file_info.src_path,
+                                        self.archive_path).replace("\\", "/")
+            result = self.compile_code(code_path)
```

Comment:
Where will the cache be written here ? To me it appears, this should lead to write the cache in the archive.